src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -181,10 +200,26 @@ private ModifiableRealmIdentity getRealmIdentity(final Principal principal, fina
     }
 
     private DirContext obtainContext() throws RealmUnavailableException {
-        try {
-            return dirContextSupplier.get();
-        } catch (NamingException e) {
-            throw log.ldapRealmFailedToObtainContext(e);
+        if (this.listenersPendingRegistration.size() == 0) {
+            try {
+                return dirContextSupplier.get();
+            } catch (NamingException e) {
+                throw log.ldapRealmFailedToObtainContext(e);
+            }
+        } else {
+            try {
+                DirContext ctx = dirContextSupplier.get();
+                synchronized (this.listenersPendingRegistration) {
+                    //we got ctx, this means connection is up
+                    for(Consumer<Principal> listener:this.listenersPendingRegistration){
```

Comment:
Minor, but when you will be already in updating, please add spaces around `:` to keep codestyle.